Source/JavaScriptCore/heap/Heap.cpp

```diff
@@ -138,6 +138,16 @@ size_t minHeapSize(HeapType heapType, size_t ramSize)
     }
 }
 
+size_t maxEdenSizeForRateLimiting(size_t ramSize, size_t minBytesPerCycle)
+{
+    // Only do rate limiting for machines >= 16GB RAM.
+    size_t aggressiveHeapThresholdInBytes = static_cast<size_t>(Options::aggressiveHeapThresholdInMB()) * MB;
+    if (ramSize < aggressiveHeapThresholdInBytes)
+        return 0;
```

Comment:
Maybe we should factor this condition check out, and place it into a member of `Heap` as enum, like,
`enum class GrowthMode : uint8_t { Default, Aggressive };`
and
`GrowthMode m_growthMode { ... }`
And passing this enum to this function instead of computing this in different places multiple times.